cmd/intervalContacts/intervalContacts.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/bed"
+	"github.com/vertgenlab/gonomics/bed/bedpe"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fileio"
+	"github.com/vertgenlab/gonomics/interval"
+	"log"
+)
+
+func intervalContacts(bedpeFile string, inFile string, contactOutFile string) {
+	var inIntervals = make([]interval.Interval, 0)
+	var currOverlaps []interval.Interval
+	var err error
+
+	out := fileio.EasyCreate(contactOutFile)
+
+	contactRecords := bedpe.Read(bedpeFile)
+	inChan := interval.GoReadToChan(inFile)
+
+	for i := range inChan {
+		inIntervals = append(inIntervals, i)
+	}
+	inTree := interval.BuildTree(inIntervals)
+
+	for _, i := range contactRecords {
+		currOverlaps = interval.Query(inTree, i.A, "any")
+		if len(currOverlaps) > 0 {
+			bed.WriteBed(out, i.B) // if anything in the input set overlaps i.A, write i.B to the output.
```

Comment:
Why do we write/output bed B when we have an overlap with bed A?

Reply:
I forget if you were there when Riley discussed his last pull request on bedpe, but I learned from there that since the bedpe files report interacting regions in the genome (e.g. Hi-C), we want to use the bedpe files to see what the regions the HAQERs are interacting with, meaning if a HAQER overlaps with A, then we are interested in what B is because B is the target of A aka the target of the HAQER which might be an enhancer